scripts/install-helpers/baremetal-coco/install.sh

```diff
@@ -0,0 +1,733 @@
+        ready=$(oc get runtimeclass "$runtimeclass" -o jsonpath='{.metadata.name}')
+        if [ "$ready" == "$runtimeclass" ]; then
+            echo "Runtimeclass $runtimeclass is ready"
+            return 0
+        fi
+        sleep $interval
+        elapsed=$((elapsed + interval))
+    done
+
+    echo "Runtimeclass $runtimeclass is not ready after $timeout seconds"
+    return 1
+}
+
+# Function to apply the operator manifests
+function apply_operator_manifests() {
+    # Apply the manifests, error exit if any of them fail
+    oc apply -f ns.yaml || exit 1
+    oc apply -f og.yaml || exit 1
+    if [[ "$GA_RELEASE" == "true" ]]; then
+        oc apply -f subs-ga.yaml || exit 1
+    else
+        oc apply -f osc_catalog.yaml || exit 1
+        oc apply -f subs.yaml || exit 1
+    fi
+
+}
+
+# Function to check if single node OpenShift
+function is_single_node_ocp() {
+    local node_count
+    node_count=$(oc get nodes --no-headers | wc -l)
+    if [ "$node_count" -eq 1 ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Function to set additional cluster-wide image pull secret
+# Requires PULL_SECRET_JSON environment variable to be set
+# eg. PULL_SECRET_JSON='{"my.registry.io": {"auth": "ABC"}}'
+function add_image_pull_secret() {
+    # Check if SECRET_JSON is set
+    if [ -z "$PULL_SECRET_JSON" ]; then
+        echo "PULL_SECRET_JSON environment variable is not set"
+        echo "example PULL_SECRET_JSON='{\"my.registry.io\": {\"auth\": \"ABC\"}}'"
+        exit 1
+    fi
+
+    # Get the existing secret
+    oc get -n openshift-config secret/pull-secret -ojson | jq -r '.data.".dockerconfigjson"' | base64 -d | jq '.' >cluster-pull-secret.json ||
+        exit 1
+
+    # Add the new secret to the existing secret
+    jq --argjson data "$PULL_SECRET_JSON" '.auths |= ($data + .)' cluster-pull-secret.json >cluster-pull-secret-mod.json || exit 1
+
+    # Set the image pull secret
+    oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=cluster-pull-secret-mod.json || exit 1
+
+}
+
+#Function to deploy the OpenShift sandboxed containers (OSC) operator
+function deploy_osc_operator() {
+    echo "OpenShift sandboxed containers operator | starting the deployment"
+    # Apply the operator manifests
+    apply_operator_manifests
+
+    wait_for_deployment controller-manager openshift-sandboxed-containers-operator || return 1
+
+    # Wait for the service endpoints IP to be available
+    wait_for_service_ep_ip webhook-service openshift-sandboxed-containers-operator || return 1
+
+    echo "OpenShift sandboxed containers operator | deployment finished successfully"
+}
+
+# Function to deploy NodeFeatureDiscovery (NFD) operator
+function deploy_nfd_operator() {
+    echo "Node Feature Discovery operator | starting the deployment"
+
+    pushd nfd || return 1
+    oc apply -f ns.yaml || return 1
+    oc apply -f og.yaml || return 1
+    oc apply -f subs.yaml || return 1
+    popd || return 1
```

Comment:
The only reason for `popd` to fail would be the removal of the current working directory at the time `pushd` was called... can this even happen ?

Reply:
This was added to keep shellcheck happy - https://www.shellcheck.net/wiki/SC2164

Reply:
The return statements for both pushd and popfd were dictated by shellcheck. I can disable it as well, but thought it's better to keep it.  Let me know if you have a strong preference to remove it :-)

Reply:
I prefer to have shellcheck happy :wink: